### PR TITLE
Pytest + Geth Fixture

### PR DESCRIPTION
**Pull request by KPrasch**

### CODE REVIEW

File: tests/eth_fixtures.py [hunk 1]
```diff
@@ -16,51 +17,83 @@
...
-    test_provider = EthereumTesterProvider(ethereum_tester=eth_tester)  # , api_endpoints=None)
+    #
+    # spin-up geth
+    #
+    testing_dir = tempfile.mkdtemp()
+    chain_name = 'nkms_tester'
+    geth = DevGethProcess(chain_name=chain_name, base_dir=testing_dir)
+    geth.start()
+
+    geth.wait_for_ipc(timeout=2)
```
Comment (michwill) [inline tests/eth_fixtures.py]: 30?

File: tests/eth_fixtures.py [hunk 2]
```diff
@@ -16,51 +17,83 @@
...
     """
     Provider backend
     https: // github.com / ethereum / eth - tester     # available-backends
     """
-    eth_tester = EthereumTester(backend=PyEVMBackend())
-    test_provider = EthereumTesterProvider(ethereum_tester=eth_tester)  # , api_endpoints=None)
+    #
+    # spin-up geth
+    #
+    testing_dir = tempfile.mkdtemp()
```
Comment (michwill) [inline tests/eth_fixtures.py]: Context manager? https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryDirectory

Comment (KPrasch) [reply tests/eth_fixtures.py]: We decided against this for better multi-platform support.

File: tests/eth_fixtures.py (hunk 1 above)
Comment (KPrasch) [reply tests/eth_fixtures.py]: Seems long! 10? (this is a module level fixture)

**Pull request merged**